src/form-engine.component.tsx

```diff
@@ -69,6 +76,10 @@ export interface FormSubmissionHandler {
   validate: (values) => boolean;
 }
 
+const functionStore = createGlobalStore<HandleConfirmDeletionFunctionStore | null>('functionStore', {
```

Comment:
I think rather than using a global store for this, it makes more sense to use a React context. Context's are local to a React tree, so we have fewer issues around naming conflicts, etc. Global stores are for things we need to be shared across different React trees, but here I don't see a reason for it.

Reply:
Thanks, Ian. I've updated to using context instead, as suggested. 
Could you please review the context part? I hope I've implemented it right.

Reply:
LGTM. I might rename the context `ExternalFunctionContext` or something so it's reusable for other similar functions (since I imagine we'll need something like this for other uses). Nice job!